plugins/inputs/snmp/snmp.go

```diff
@@ -97,24 +116,32 @@ type Snmp struct {
 	connectionCache []snmpConnection
 
 	Log telegraf.Logger `toml:"-"`
+
+	translator Translator
 }
 
 func (s *Snmp) Init() error {
-	err := snmp.LoadMibsFromPath(s.Path, s.Log, &snmp.GosmiMibLoader{})
-	if err != nil {
-		return err
+	var err error
+
+	if s.Translator == "gosmi" {
+		s.translator, err = NewGosmiTranslator(s.Path, s.Log)
+		if err != nil {
+			return err
+		}
+	} else {
+		s.translator = NewNetsnmpTranslator()
 	}
```

Comment:
This could maybe better done in a separate method? `func (s *Snmp) initTranslator() error {` Then this `Init` method could return error `fmt.Errorf("initialising translator %s: %w", s.Translator, err)`

Reply:
It's only done once so I don't think it needs a separate method. There's a similar pattern in snmp_trap.go too, but it uses a different interface.